### PR TITLE
[python] implement str.join() method with string concatenation

### LINKED ISSUE #3000: [Python] Crash with join operation
Program:
```python
l: list[str] = ["foo", "bar", "baz"]
s = " ".join(l)
assert s == "foo bar baz"
```
ESBMC output:
```
$ esbmc ex.py 
ESBMC version 7.11.0 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing ex.py
terminate called after throwing an instance of 'nlohmann::json_abi_v3_11_3::detail::type_error'
  what():  [json.exception.type_error.302] type must be string, but is number
Aborted (core dumped)
```

### PR DESCRIPTION
**Pull request by LukeW1999**
Fixes https://github.com/esbmc/esbmc/issues/3000.

## In short
Implements support for Python's `str.join()` method in the ESBMC Python frontend.

## Implementation
- Added `handle_str_join()` method in `python_converter.cpp`
- Fixed crash when calling methods on string literals in `python_annotation.h`
- Supports variable references, empty lists, single elements, and multi-character separators

## Test Coverage
Added 17 regression tests covering:
- Basic join operations 
- Edge cases (empty list, single element, empty separator)
- Type error test (`github_3000_8_fail`)

## Known Issues

**Limited support for direct list literals:**
`github_3000_1` is marked as `KNOWNBUG` - currently optimized for variable references. It should be able to set string as default and do a check in future since join's nature of string elements.

**Counterexample readability:**
When verification fails, the output shows low-level C symbols instead of Python-level information. For example, `github_3000_7_fail` outputs:
```
Violated property: return_value$_strcmp$1 == 0
```
Instead of showing the actual vs expected string values.

### CODE REVIEW

**Review by lucasccordeiro — APPROVED** [review]
Thanks for submitting this PR, @LukeW1999.

File: src/python-frontend/python_annotation.h [hunk 1]
```diff
@@ -1139,6 +1139,25 @@ class python_annotation
...
+    else if (value.is_number_integer())
+      return "int";
+    else if (value.is_number_float())
+      return "float";
+    else if (value.is_boolean())
+      return "bool";
+
+    return "";
+  }
+
```
Comment (brcfarias) [inline src/python-frontend/python_annotation.h:1160]: @LukeW1999,
The `get_type_from_constant` function already existed in this file. It's duplicated now:
https://github.com/esbmc/esbmc/blob/master/src/python-frontend/python_annotation.h#L589-L607
https://github.com/esbmc/esbmc/blob/master/src/python-frontend/python_annotation.h#L1143-L1159
Could you please remove one of them?

Comment (LukeW1999) [reply src/python-frontend/python_annotation.h:1160]: Hi, @brcfarias 
Sure. I would fix it when I went back to home. 
I am really sorry for my mistake. Thanks a lot for telling me.

Comment (brcfarias) [reply src/python-frontend/python_annotation.h:1160]: That’s fine. I don’t think it’s urgent :-)
